### PR TITLE
Replace deprecated APIs from FairMQ and Geant4

### PR DESCRIPTION
**Pull request by karabowi**
Addresses issue #1501 in FairMQ part.
Also Geant4 deprecates functions like Set...() in favor of Propose...().

---

Checklist:

* [x] Followed the [Contributing Guidelines](https://github.com/FairRootGroup/FairRoot/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

File: examples/MQ/histogramServer/FairMQExHistoDevice.cxx [hunk 1]
```diff
@@ -55,7 +55,7 @@ bool FairMQExHistoDevice::ConditionalRun()
     auto message(NewMessage());
     RootSerializer().Serialize(*message, &fArrayHisto);
 
-    for (auto& channel : fChannels) {
+    for (auto& channel : GetChannels()) {
         Send(message, channel.first);
```
Comment (dennisklein) [inline examples/MQ/histogramServer/FairMQExHistoDevice.cxx:59]: Using the new API unconditionally means you require FairMQ v1.5.0. Is this intended? If yes, then we should express this in the `find_package` call.

**Comment by rbx** [comment]
I attempted to test this on my machine, but I get the same compilation error as on cdash: https://cdash.gsi.de/viewBuildError.php?buildid=447412
With FairSoft jan24.

**Comment by rbx** [comment]
As for the FairMQ-related changes, they look good to me and get rid of the warnings. Indeed, as Dennis said, `GetChannels` is only available since FairMQ v1.5.0, so that should be bumped.

**Comment by dennisklein** [comment]
> As for the FairMQ-related changes, they look good to me and get rid of the warnings. Indeed, as Dennis said, `GetChannels` is only available since FairMQ v1.5.0, so that should be bumped.

Whether to bump the required FairMQ version should depend on the range of FairSoft versions intended to be supported!

**Comment by karabowi** [comment]
You are all right. Since we want to have the v19 also working with nov22, we should not bump the FairMQ version, but rather deal with the deprecation warning other way. Closing this PR and waiting for Dennis to deal with issue https://github.com/FairRootGroup/FairRoot/issues/1501.

**Comment by dennisklein** [comment]
Btw, the analog question is also open regarding the minimum required ROOT version. Since we switch to discover ROOT via its CMake package, we implicitely bump the min version to the first ROOT release that ships a CMake package. However, in our CMake we also state some version.

In general, the version combinatorics are poorly managed in FairRoot, imho we should improve here and clearly document which versions FairRoot intends to support (for all deps). Not just in terms of FairSoft, technically, FairRoot should not need to know about FairSoft, meaning we should resolve the required versions (which may be defined via FairSoft) to a per package term. At least, if we ever want to achieve regular packability of FairRoot.

**Pull request closed without merging**